### PR TITLE
Add Docker EE support for Windows Server 2016

### PR DESCRIPTION
**Pull request by mihaibuzgau**
This PR adds the functionality to deploy Docker Enterprise Edition on Windows Server 2016. The deployment is done using the Docker module available in the online Powershell gallery.

This PR also adds the functionality to deploy Docker images on Windows.

### CODE REVIEW

**Comment by davejrt** [comment]
@mihaibuzgau this is a great start. Let's work on getting the unit and acceptance tests completed before we merge this into master though.

**Comment by mihaibuzgau** [comment]
@davejrt agreed. I'll add some unit and acceptance tests as soon as I get the chance.

File: templates/windows/config/daemon.json.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+    <%# TODO: get "authorization-plugins": [],%>
```
Comment (davejrt) [inline templates/windows/config/daemon.json.erb]: Can we remove any commented code, just so there's nothing WIP sitting in master?

Comment (mihaibuzgau) [reply templates/windows/config/daemon.json.erb]: sure

**Comment by davejrt** [comment]
LGTM

**Pull request merged**